### ISSUE TITLE
[OJS] upgrading to OJS3: Revisions and Reviewer's Attachments panels not visible in Review tab for authors

### ISSUE THREAD

**Issue by ajnyga** [pkp/pkp-lib#2159]
After upgrading from OJS2 to OJS3.0.1, the authors can not see the Revisions and Reviewer's Attachments panels when they go to the Review tab in their submissions. Only the discussion panel is visible. If there is a decision done with the submission, it is visible in the discussion panel with a title Editor Decision, but the two other panels are still missing.

If I make a new submission and initate the review phase, all the panels are visible as expected. So this is a migration problem. 

![screenshot_48](https://cloud.githubusercontent.com/assets/16347527/21742856/c389830e-d4ff-11e6-9945-57b33560b841.png)

**Comment by ajnyga** [comment]
The reason seems to be that in edit_decisions table the migrated submissions do not have review_round_id or stage_id set:
![screenshot_49](https://cloud.githubusercontent.com/assets/16347527/21742975/1a3e0e8e-d502-11e6-8579-7c58b89c2c61.png)

After adding those, the two missing panels appear.

**Comment by ajnyga** [comment]
Another issue with these old submissions is that if the author tries to upload new files for example in the discussion section there is a "HTTP error". You do not get that error with new submissions. Also the fix mentioned aboce does not solve the upload issue - the panel for the revisions appears, but the author can not upload anything.

So there is other migration issues here as well.

**Comment by ajnyga** [comment]
Actually setting the stage_id is enough.

**Comment by ajnyga** [comment]
Basically `showReviewAttachments` is not getting value "true" here: https://github.com/pkp/pkp-lib/blob/master/controllers/tab/authorDashboard/AuthorDashboardReviewRoundTabHandler.inc.php because `getEditorTakenActionInReviewRound` is not returning anything here: https://github.com/pkp/ojs/blob/master/classes/workflow/EditorDecisionActionsManager.inc.php

**Comment by bozana** [comment]
I will take a look...

**Comment by ajnyga** [comment]
After chancing the stage_id the two panels are visible.

However, the file in the revision grid can not be accessed:
{"status":false,"content":"The current role does not have access to this operation.","elementId":"0"}

**Comment by ajnyga** [comment]
@bozana or @asmecher, is there logic there I can use to fill the missing stage_id fields? Is it always 3?

**Comment by ajnyga** [comment]
I added a separate issue concerning the file access error in the uploaded files.

**Comment by bozana** [comment]
Concerning the missing stage_id:
If I see it correctly, in the `editor_decisions` table in OJS 2.4.x there were only entries for the review stage, right @ajnyga? Thus, yes, for all submission_ids from the OJS 2.4.x you could add the `stage_id = 3` there. If you would like, you could send me the DB dump of that table and I could double check it...

**Comment by bozana** [comment]
@ajnyga, what does this mean: 

> Also, there are many cases where there is no entry at all in edit_decisions table, although the review has started. Any idea if this can be fixed in the upgrade script?

Do you mean no entry for stage_id? Or do you mean no entries for other stages?

**Comment by ajnyga** [comment]
Thanks @bozana, I will check these in a few hours.

**Comment by bozana** [comment]
PRs:
ojs-stable-3_0_1: https://github.com/pkp/ojs/pull/1188

**Comment by bozana** [comment]
@asmecher, could you review the PR above?

Also, do you think we should insert the review_round_id where possible, with:
`UPDATE edit_decisions ed, review_rounds rr SET ed.review_round_id = rr.review_round_id WHERE rr.round = ed.round AND rr.submission_id = ed.submission_id
`

And also, do you think we should insert the extra rows for submission and editing stage for each submission, as it is always in OJS 3.x? -- Their missing seems to make no problems for now...

**Comment by ajnyga** [comment]
Thank you @bozana,

I checked the edit_decision table from the old OJS2 database and yes, there are only decisions 1, 2,  3, 4 there. 

In the upgraded OJS3 table there are also other decisions, but those are there only after the upgrade.

Concerning the other case, I meant that whole rows are missing from the table in some cases where an article is in review. However, naturally I could not find a similar case now. I will get back to you when I do.

**Comment by asmecher** [comment]
> And also, do you think we should insert the extra rows for submission and editing stage for each submission, as it is always in OJS 3.x? -- Their missing seems to make no problems for now...

Sorry, which rows?

**Comment by bozana** [comment]
@asmecher, I mean rows in the table edit_decisions for the next stage 4. They are not there after the upgrade. Also the rows for the stage = 1 are not there. I.e. only review stage rows are in the table edit_decisions, which makes it easy to fix the bug here :-)

**Comment by bozana** [comment]
@asmecher, there is one more thing I noticed: in OJS 2.4.x review_rounds table contain entries only if a review took place, while in OJS 3.0 there is always an entry in the review_rounds table, also when an editor just clicks "Send To Copyediting" (and do not involve any reviewers). This means that there could be entries with review_round_id = NULL in the table edit_decisions after the migration. What to do about this? -- Either to leave it as it is, or to try to fill the review_rounds table accordingly, and then also the missing review_round_ids in the edit_decisions table?

**Comment by asmecher** [comment]
Ah, I see. I'd suggest adding them -- it may not matter now but consistency is a good thing for 3 years from now :)

**Comment by bozana** [comment]
@asmecher, could we maybe split this issue and first do this fix and then the others: https://github.com/pkp/ojs/pull/1188 -- I need time to figure out how to fill the edit_decisions with missing rows, to consider all cases... If so, could you please review the PR again? -- I considered the changes for other OJS versions...

**Comment by asmecher** [comment]
Sure. The PR looks good, please go ahead and merge/forward-port!

**Comment by asmecher** [comment]
(The remaining piece might be one of those things that gets done in a PHP loop rather than a clever bit of SQL...)

**Comment by bozana** [comment]
cosmetic PRs:
ojs-stable-3_0_2: https://github.com/pkp/ojs/pull/1319

**Comment by bozana** [comment]
@asmecher, could you please take a look at this last PR? -- it is just a cosmetic fix...

**Comment by asmecher** [comment]
Merged it, thanks!

**Comment by asmecher** [comment]
Can this now be closed?

**Comment by bozana** [comment]
everything merged, thus closing...

**Comment by kerimsarigul** [comment]
I am facing exactly this issue right now.
Revisions and Reviewer's Attachments panels not visible in Review tab, as @ajnyga stated at the top of the page.
I couldn't find a way to fix this problem.
![image](https://user-images.githubusercontent.com/54417583/164531359-ec4a1e4b-b14c-40a9-9b59-4acdd5c0bb76.png)


![image](https://user-images.githubusercontent.com/54417583/164530520-34dc76ec-a9b6-40c5-b7dd-6c6cdb7f22d5.png)

![image](https://user-images.githubusercontent.com/54417583/164532284-e5e3171f-f7b2-4e15-80b4-9640e79fc472.png)

**Comment by asmecher** [comment]
@kerimsarigul, I see you've also posted on the support forum:

https://forum.pkp.sfu.ca/t/revisions-and-reviewers-attachments-panels-not-visible-in-review-tab-for-authors/72939

That's the best place to get help; in general it's best to just post your question in one place.
